mozjs-sys/src/jsgc.rs

```diff
@@ -111,14 +111,12 @@ pub unsafe trait TraceableTrace: Sized {
     /// Used by `TraceableTrace` implementer to trace its contents.
     /// Corresponds to virtual `trace` call in a `Rooted` that inherits from
     /// StackRootedTraceableBase (C++).
-    fn do_trace(&mut self, trc: *mut JSTracer);
+    unsafe fn do_trace(&mut self, trc: *mut JSTracer);
 }
 
 unsafe impl TraceableTrace for JS::PropertyDescriptor {
-    fn do_trace(&mut self, trc: *mut JSTracer) {
-        unsafe {
-            CallPropertyDescriptorTracer(trc, self);
-        }
+    unsafe fn do_trace(&mut self, trc: *mut JSTracer) {
+        CallPropertyDescriptorTracer(trc, self);
     }
```

Comment:
Actually unsafe block in unsafe fn will be the only correct way in the future: https://doc.rust-lang.org/edition-guide/rust-2024/unsafe-op-in-unsafe-fn.html, but that is not yet relevant.